File: app/src/main/java/org/akvo/flow/presentation/navigation/ViewSurvey.java
/*
 * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
 *
 * This file is part of Akvo Flow.
 *
 * Akvo Flow is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Akvo Flow is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.akvo.flow.presentation.navigation;

import android.os.Parcel;
import android.os.Parcelable;

class ViewSurvey implements Parcelable {

    private final long id;
    private final String name;
    private final boolean isMonitored;
    private final String registrationSurveyId;

    ViewSurvey(long id, String name, boolean isMonitored, String registrationSurveyId) {
        this.id = id;
        this.name = name;
        this.isMonitored = isMonitored;
        this.registrationSurveyId = registrationSurveyId;
    }

    ViewSurvey(Parcel in) {
        id = in.readLong();
        name = in.readString();
        isMonitored = in.readByte() != 0;
        registrationSurveyId = in.readString();
    }

    public static final Creator<ViewSurvey> CREATOR = new Creator<ViewSurvey>() {
        @Override
        public ViewSurvey createFromParcel(Parcel in) {
            return new ViewSurvey(in);
        }

        @Override
        public ViewSurvey[] newArray(int size) {
            return new ViewSurvey[size];
        }
    };

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public boolean isMonitored() {
        return isMonitored;
    }

    public String getRegistrationSurveyId() {
        return registrationSurveyId;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(id);
        dest.writeString(name);
        dest.writeByte((byte) (isMonitored ? 1 : 0));
        dest.writeString(registrationSurveyId);
    }
}

File: app/src/main/java/org/akvo/flow/presentation/navigation/ViewSurvey.kt
/*
 * Copyright (C) 2017,2020 Stichting Akvo (Akvo Foundation)
 *
 * This file is part of Akvo Flow.
 *
 * Akvo Flow is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Akvo Flow is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package org.akvo.flow.presentation.navigation

import android.os.Parcel
import android.os.Parcelable

data class ViewSurvey(
    val id: Long,
    val name: String?,
    val isMonitored: Boolean,
    val registrationSurveyId: String?,
    val viewed: Boolean
) : Parcelable {

    constructor(parcel: Parcel) : this(
        parcel.readLong(),
        parcel.readString(),
        parcel.readByte() != 0.toByte(),
        parcel.readString(),
        parcel.readByte() != 0.toByte()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id)
        parcel.writeString(name)
        parcel.writeByte(if (isMonitored) 1 else 0)
        parcel.writeString(registrationSurveyId)
        parcel.writeByte(if (viewed) 1 else 0)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ViewSurvey> {
        override fun createFromParcel(parcel: Parcel): ViewSurvey {
            return ViewSurvey(parcel)
        }

        override fun newArray(size: Int): Array<ViewSurvey?> {
            return arrayOfNulls(size)
        }
    }
}
